domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/timeline/TimelineEvent.kt

```diff
@@ -23,9 +23,9 @@ data class TimelineEvent(
    */
   val eventType: TimelineEventType,
   /**
-   * Some useful contextual information about the event. For example a reason for the event.
+   * A map of useful contextual information about the event.
    */
-  val contextualInfo: String?,
+  val contextualInfo: Map<TimelineEventContext, String>?,
```

Comment:
We are changing `contextualInfo` from being a simple string to a `Map<TimelineEventContext, String>`, where `TimelineEventContext` is an enum of values (defined below)

This is the domain class; the rest of the changes in the PR relate to the database, API and mappers etc where we are making essentially the same changes